### PR TITLE
Fabo/extract signer libs

### PR DESCRIPTION
**Pull request by faboweb**
Closes #ISSUE

**Description:**

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: src/scripts/ledger.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  async testDevice(outdated) {
+    // poll device with low timeout to check if the device is connected
+    const secondsTimeout = 3 // a lower value always timeouts
+    await this.connect(secondsTimeout, outdated)
+  }
+  async isSendingData() {
+    // check if the device is connected or on screensaver mode
+    const response = await this.cosmosApp.publicKey(HDPATH)
+    this.checkLedgerErrors(response, {
+      timeoutMessag: "Could not find a connected and unlocked Ledger device"
```
Comment (colw) [inline src/scripts/ledger.js]:
```suggestion
      timeoutMessage: "Could not find a connected and unlocked Ledger device"
```

File: src/vuex/modules/tx.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
+{
```
Comment (colw) [inline src/vuex/modules/tx.json]: Where is this used? I'm probably being blind here

Comment (faboweb) [reply src/vuex/modules/tx.json]: that was a demo for all the tx types. somehow ended up there. I removed it.

File: src/components/wallet/SendModal.vue [hunk 3]
```diff
@@ -187,23 +187,29 @@ export default {
...
       const address = this.address
       const denom = this.denom
-      const type = `send`
+      const type = `MsgSend`
 
       return await this.$store.dispatch(`simulateTx`, {
         type,
-        to: address,
-        amount: [{ denom, amount: String(uatoms(amount)) }],
+        txArguments: {
```
[the remark this answers is not in the document]
Comment (faboweb) [reply src/components/wallet/SendModal.vue]: As I want to get rid of those two actions and move them into the ActionModal, I would like to not do this refactor in this PR if that'S ok.

[the remark this answers is not in the document]
Comment (colw) [reply src/components/wallet/SendModal.vue]: Of course, makes sense.

**Pull request merged**